cfme/tests/infrastructure/test_vmrc_console.py

```diff
@@ -2,19 +2,63 @@
 
 from cfme import test_requirements
 from cfme.infrastructure.provider.virtualcenter import VMwareProvider
+from cfme.utils.log import logger
 
 
-pytestmark = [
-    pytest.mark.usefixtures('setup_provider_modscope'),
-    pytest.mark.provider([VMwareProvider], scope='module')
-]
+@test_requirements.vmrc
+@pytest.mark.provider([VMwareProvider],
+    required_fields=[(['cap_and_util', 'capandu_vm'], 'cu-24x7')])
+def test_vmrc_console(request, appliance, setup_provider, provider, configure_console_vmrc):
+    """
+    Test VMRC console can be opened for the VMware provider.
+
+    Polarion:
+        assignee: apagac
+        casecomponent: Infra
+        caseimportance: high
+        initialEstimate: 2h
+        setup:
+            1. Login to CFME Appliance as admin.
+            2. Navigate to Configuration
+            3. Under VMware Console Support section and click on Dropdown in front
+               of "Use" & select "VMware VMRC Plugin"(Applicable to versions older than CFME 5.11).
+            4. Click save at the bottom of the page.
+            5. Provision a testing VM.
+        testSteps:
+            1. Navigtate to testing VM
+            2. Launch the console by Access -> VM Console
+        expectedResults:
+            1. VM Details displayed
+            2. You should see a Pop-up being Blocked, Please allow it to
+               open (always allow pop-ups for this site) and then a new tab
+               will open and then in few seconds for the VMRC console.
+    """
+    vms_collections = appliance.collections.infra_vms
+    vm = vms_collections.instantiate(name='cu-24x7', provider=provider)
+    if not vm.exists_on_provider:
+        pytest.skip("Skipping test, cu-24x7 VM does not exist")
+    if appliance.version < '5.11':
+        try:
+            vm.open_console(console='VM Console', invokes_alert=True)
+            assert vm.vm_console, 'VMConsole object should be created'
+            request.addfinalizer(vm.vm_console.close_console_window)
+        except ValueError:
```

Comment:
If the VM console fails to launch here, we are giving a fast positive. Going to discuss a workaround offline on Monday

Reply:
Is the window handle different in 5.10 vs 5.11? Just a thought.

Reply:
I think Window Handles are Selenium thing and should have nothing to do with CFME version. Good question though, I will checl.